Separate preview volume

maks opened the pull request:
This provides for a separate volume setting for sample preview volume and provides a UI for adjusting it on the import screen using `ENTER+UP/DOWN`

Fixes: #487

democloid commented:
@maks This is not working for me, not sure what's going on, but EDIT+UP/DOWN does nothing


I'm wondering if we should:
1. generalize to have the same keycombo to adjust master volume outside of sample preview
2. have a better UI, status bar probably
3. Is this the right keycombo? seems ok, but I'm getting a bit lost by all the new functionality, feel we should have some broader view to all the combos we have

maks commented:
> @maks This is not working for me, not sure what's going on, but EDIT+UP/DOWN does nothing

oops sorry that was a typo in the PR description I just fixed, its **ENTER** not EDIT.
> 
> I'm wondering if we should:
> 
> 1. generalize to have the same keycombo to adjust master volume outside of sample preview

Good point! I'll need to just check that the keycombo is available on all screens and all fields in screens, but this should be a separate issue.

> 2. have a better UI, status bar probably

For sure, though thats a really separate issue too.

> 3. Is this the right keycombo? seems ok, but I'm getting a bit lost by all the new functionality, feel we should have some broader view to all the combos we have

Well it is all documented in the user manual, though for this I only added it to the relevant chapter (instruments) but good point that it needs to be added to the central key combo list too 👍🏻

democloid reviewed (commented):
Code is fine. UI I think need some adjustments/discussion:
1. So this is using a log scale, which is fine, but doesn't align with master volume, no? Also master volume goes from 0 - 255, while this goes from 0% - 100%. I think we should decide on one way to do volume and be consistent
2. The "restart if volume is adjusted" I don't think I like. I don't have a long sample to test at hand, but I'm I'm changing volume I probably don't want the sample to restart with each keypress. start/stop is easy enough that I can do if I want. And in most cases samples are short enough that this doesn't do anything useful anyway. Thoughts?
3. I meant volume in title bar rather than status, that's on me. But we need to figure out layout

maks commented:
> Code is fine. UI I think need some adjustments/discussion:
> 1. So this is using a log scale, which is fine, but doesn't align with master volume, no? Also master volume goes from 0 - 255, while this goes from 0% - 100%. I think we should decide on one way to do volume and be consistent

No, master volume doesn't work that way any more either but that was a separate issue so done in another another in-review PR.


> 2. The "restart if volume is adjusted" I don't think I like. I don't have a long sample to test at hand, but I'm I'm changing volume I probably don't want the sample to restart with each keypress. start/stop is easy enough that I can do if I want. And in most cases samples are short enough that this doesn't do anything useful anyway. Thoughts?

Hmm I'll need to have a think about this one.

> 3. I meant volume in title bar rather than status, that's on me. But we need to figure out layout

Haha ok, I thought you meant statusbar. Can definitely do in top row of course but TBH I think it makes just as much sense to keep it on bottom with the existing statusbar, thoughts?

democloid reviewed (approved):
This is fine but I strongly believe we should remove the "restart sample playback when volume is adjusted", it's not something you generally want IMO. Say we have the sequencer on and you change master volume, you don't restart because you change the volume. I don't think this adds much and it makes code more complicated

maks commented:
> This is fine but I strongly believe we should remove the "restart sample playback when volume is adjusted", it's not something you generally want IMO. Say we have the sequencer on and you change master volume, you don't restart because you change the volume. I don't think this adds much and it makes code more complicated

Yes good point! I think this is a hangover from when previously we had preview start/stop via PLAY key vs how it is now changed to keeping PLAY pressed to keep sample preview playing, so yep, removed this restart code 👍🏻 
But the restart will still actually happen when pressing any other key while holding the PLAY key just because of how the key event handling is currently implemented on the import subscreen but I think thats a minor issue we can address later if needs be.
